akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala

```diff
@@ -477,8 +461,13 @@ private[akka] class RemoteActorRef private[akka] (
 
   def sendSystemMessage(message: SystemMessage): Unit =
     try {
-      remote.send(message, None, this)
-      provider.afterSendSystemMessage(message)
+      message match {
+        case Watch(watchee, watcher) if watcher != provider.remoteWatcher ⇒
+          provider.remoteWatcher ! RemoteWatcher.WatchRemote(watchee, watcher)
```

Comment:
yes, this is the idea of the change: only watches from the remoteWatcher are sent to the remote side